src/sentry/mediators/external_requests/util.py

```diff
@@ -53,24 +53,45 @@ def send_and_save_sentry_app_request(url, sentry_app, org_id, event, **kwargs):
     buffer = SentryAppWebhookRequestsBuffer(sentry_app)
 
     slug = sentry_app.slug_for_metrics
+    error_id = None
+    project_id = None
 
     try:
         resp = safe_urlopen(url=url, **kwargs)
-    except RequestException:
+        error_id = resp.headers.get("Sentry-Hook-Error")
+        project_id = resp.headers.get("Sentry-Hook-Project")
```

Comment:
@NisanthanNanthakumar Why not just call `track_response_code` and `buffer.add_request` here instead?

Reply:
@scefali I was not sure if calling `resp.status_code` for a `timeout` would cause a different exception. Just tested it out and it doesn't blow up. I'll make the updates.

Reply:
@NisanthanNanthakumar It would. But the timeout will error out when `safe_urlopen` so it should never get to that line.